pkg/cfn/builder/templates/beta.py

```diff
@@ -0,0 +1,405 @@
+    eks_client.delete_cluster(name=cluster_name)
+    logger.info(f"EKS cluster deleted: {cluster_name}")
+
+
+def nodegroup_handler(event, context):
+    """
+    Handles the creation, update, and deletion of an EKS managed node group as a CloudFormation custom resource.
+    """
+    try:
+        eks_client = init_eks_client()
+        iam_client = boto3.client('iam')
+
+        cluster_name = event['ResourceProperties']['ClusterName']
+        logger.info(f"cluster name : {cluster_name}")
+
+        # Handle Delete event
+        if event['RequestType'] == 'Delete':
+            nodegroup_name = event['ResourceProperties']['NodegroupName']
+            logger.info(f"nodegroup name : {nodegroup_name}")
+
+            response = eks_client.describe_nodegroup(
+                clusterName=cluster_name,
+                nodegroupName=nodegroup_name
+            )
+            nodegroup_role_arn = response['nodegroup']['nodeRole']
+            print(f"IAM Role ARN associated with the EKS node group: {nodegroup_role_arn}")
+
+            role_name = nodegroup_role_arn.split('/')[-1]
+            response = iam_client.list_instance_profiles_for_role(RoleName=role_name)
+            instance_profiles = response['InstanceProfiles']
+            for profile in instance_profiles:
+                # Process each profile
+                instance_profile_name = profile['InstanceProfileName']
+                logger.info(f"Processing instance profile: {instance_profile_name}")
+
+                iam_client.remove_role_from_instance_profile(
+                    InstanceProfileName=instance_profile_name,
+                    RoleName=role_name
+                )
+                print(f"Removed role {role_name} from instance profile: {instance_profile_name}")
+
+            eks_client.delete_nodegroup(clusterName=cluster_name, nodegroupName=nodegroup_name)
+            cfnresponse.send(event, context, cfnresponse.SUCCESS, {"Message": "Resource deleted"})
+            return {
+                'PhysicalResourceId': event['PhysicalResourceId']
+            }
+
+        # Prepare the nodegroup request payload from the custom resource properties
+        nodegroup_payload = convert_keys_to_lowercase_first_letter(
+            copy.deepcopy(event['ResourceProperties']))
+        del nodegroup_payload['serviceToken']
+        replace_boolean_strings(nodegroup_payload)
+        replace_integer_strings(nodegroup_payload)
+
+        update_payload_tags(nodegroup_payload, event)
+
+        logger.info("EKS nodegroup with payload: " + json.dumps(nodegroup_payload, default=str))
+        response = eks_client.create_nodegroup(**nodegroup_payload)
+        logger.info("EKS nodegroup created: " + json.dumps(response, default=str))
+
+        eventData = {
+            "Arn": response['nodegroup']['nodegroupArn'],
+            "PhysicalResourceId": response['nodegroup']['nodegroupArn'],
+        }
+
+        cfnresponse.send(event, context, cfnresponse.SUCCESS,
```

Comment:
Is there a reason we aren't waiting for the nodegroup to become healthy here first before returning a success? I believe that's the default behavior of the official cfn handler. Not sure the implications of changing that behavior in regards to usage with eksctl.

Reply:
ah i missed that. will fix.

Reply:
Done!